filesystem/fs_interface.go

```diff
@@ -0,0 +1,14 @@
+package filesystem
+
+type Client interface {
+	Open() File
```

Comment:
the filesystem/client interface might be :

OpenReadCloser(path string) ReadCloser
OpenWriteCloser(path string) WriteCloser
Exists(path string) bool
Rename(oldPath, newPath string) error
Glob(pattern string) []string


We do not need a File interface now.

both ReadCloser and WriteCloser are defined in io package.


Xiaoyun

On Sun, Mar 8, 2015 at 5:51 AM, Hongchao Deng <notifications@github.com>
wrote:

> In filesystem/fs_interface.go
> <https://github.com/taskgraph/taskgraph/pull/18#discussion_r25999936>:
>
> > @@ -0,0 +1,14 @@
> > +package filesystem
> > +
> > +type Client interface {
> > +	Open() File
>
> Any suggestion? ReadCloser doesn't include write() and sync().
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/taskgraph/taskgraph/pull/18/files#r25999936>.
>